### ISSUE TITLE
Paste wsgi.lint finds a problem with web2 wsgi implementation.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2166]
|<img alt="martinpaljak's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| martinpaljak reported|
|-|-|
|Trac ID|trac#2166|
|Type|defect|
|Created|2006-10-11 05:24:35Z|

AssertionError: wsgi.input (<twisted.web2.wsgi.InputStream object at 0x40d73c0c doesn't have the attribute !__iter!__

The following patch fixes this:

```
--- twisted/web2/wsgi.py        (revision 18418)
+++ twisted/web2/wsgi.py        (working copy)
@@ -58,6 +58,13 @@
     def __init__(self, newstream):
         # Called in IO thread
         self.stream = stream.BufferedStream(newstream)
+    
+    def __iter__(self):
+        while 1:
+            line = self.readline()
+            if not line:
+                return
+            yield line
         
     def read(self, size=None):
         # Called in application thread

```

<details><summary>Searchable metadata</summary>

```
trac-id__2166 2166
type__defect defect
reporter__martinpaljak martinpaljak
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__ 
time__1160544275000000 1160544275000000
changetime__1160882895000000 1160882895000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)<a name="note_2"></a>|@jyknight set status to `assigned`|
|-|-|

Changes to stream.readline to fix wsgi.input's readline, readlines, and __iter__ methods:

 * Remove BufferedStream.readline maxLength argument.
 * Add size argument to wsgi.input's readline(). This is not part of the WSGI spec, but code
   (such as python stdlib cgi.py require it anyways.) ([#11006](https://github.com/twisted/twisted/issues/11006))
 * Add BufferedStream.readline size argument, to enable above.

 * Fix wsgi.readlines() and wsgi.readline() to not add extraneous delimiters. ([#2170](https://github.com/twisted/twisted/issues/2170))
 * Make BufferedStream.readline return delimiter as part of the result, to enable above.

**Comment by twisted-trac** [comment]
|<img alt="foom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|foom set status to `closed`|
|-|-|

(In [18460]) Fix wsgi.input's readline, readlines, and __iter__ methods. In detail:

 * Remove BufferedStream.readline maxLength argument.
 * Add size argument to wsgi.input's readline(). This is not part of the WSGI spec, but code
   such as python stdlib cgi.py require it anyways. ([#11006](https://github.com/twisted/twisted/issues/11006))
 * Add BufferedStream.readline size argument, to enable above.

 * Fix wsgi.readlines() and wsgi.readline() to not add extraneous delimiters. ([[#2170](https://github.com/twisted/twisted/issues/2170)](https://github.com/twisted/twisted/issues/2170))
 * Make BufferedStream.readline return delimiter as part of the result, to enable above.

 * Add an __iter__ method to wsgi.input, which is required by the spec. ([[#2166](https://github.com/twisted/twisted/issues/2166)](https://github.com/twisted/twisted/issues/2166))

 * Change fileupload's use of BufferedStream.readline to conform with above changes.

Also, add some docstrings.

Author: mkerrin, jknight
Reviewer: exarkun
Merges branch: readline-1451
Fixes [#11006](https://github.com/twisted/twisted/issues/11006)
Fixes #2166
Fixes #2170

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
